python/lsst/ctrl/mpexec/singleQuantumExecutor.py

```diff
@@ -55,24 +63,35 @@ class SingleQuantumExecutor(QuantumExecutor):
     taskFactory : `~lsst.pipe.base.TaskFactory`
         Instance of a task factory.
     skipExisting : `bool`, optional
-        If True then quanta with all existing outputs are not executed.
-    clobberPartialOutputs : `bool`, optional
-        If True then delete any partial outputs from quantum execution. If
-        complete outputs exists then exception is raise if ``skipExisting`` is
-        False.
+        If `True`, then quanta that succeeded will not be rerun.
+    clobberOutputs : `bool`, optional
+        If `True`, then existing outputs will be overwritten.  If
+        `skipExisting` is also `True`, only outputs from failed quanta will
+        be overwritten.
     enableLsstDebug : `bool`, optional
         Enable debugging with ``lsstDebug`` facility for a task.
+    exitOnKnownError : `bool`, optional
+        If `True`, call `sys.exit` with the appropriate exit code for special
+        known exceptions, after printing a traceback, instead of letting the
+        exception propagate up to calling.  This is always the behavior for
+        InvalidQuantumError.
     """
-    def __init__(self, taskFactory, skipExisting=False, clobberPartialOutputs=False, enableLsstDebug=False):
+    def __init__(self, taskFactory, skipExisting=False, clobberOutputs=False, enableLsstDebug=False,
+                 exitOnKnownError=False):
         self.taskFactory = taskFactory
         self.skipExisting = skipExisting
         self.enableLsstDebug = enableLsstDebug
-        self.clobberPartialOutputs = clobberPartialOutputs
+        self.clobberOutputs = clobberOutputs
+        self.exitOnKnownError = exitOnKnownError
 
     def execute(self, taskDef, quantum, butler):
 
         startTime = time.time()
 
+        # Save detailed resource usage before task start to metadata.
+        quantumMetadata = PropertyList()
```

Comment:
This does not need to be a property list does it?

Reply:
I'm just doing exactly what `Task` itself does to avoid surprises.